canvas_sdk/protocols/base.py

```diff
@@ -1,7 +1,17 @@
+from typing import Any
+
+from canvas_sdk.data.client import GQL_CLIENT
 from canvas_sdk.handlers.base import BaseHandler
 
 
 class BaseProtocol(BaseHandler):
     """
     The class that protocols inherit from.
     """
+
+    def run_gql_query(self, query: str, variables: dict | None = None) -> dict[str, Any]:
```

Comment:
Invoking name mangling with double underscores here might be good, which also prevents it from showing up in an IDE (PyCharm at least):

```suggestion
    def __run_gql_query(self, query: str, variables: dict | None = None) -> dict[str, Any]:
```

And calling it:

```python
protocol._BaseProtocol__run_gql_query(query, variables)
```

Reply:
gonna call it `_beta_run_gql_query` just to avoid having to put the base class in the method call